### DISCUSSION TITLE
Local matomo tracking throughout sales process happening on pretix site

### DISCUSSION THREAD

**Discussion by dune73** [pretix/pretix#4073] [Q&A] [answered]
We're running a conference and we track the visitors on our website with a local matomo (formerly piwik) instance.

We see when they click the "buy ticket" button and head over to pretix.eu and we lose track of them.

I see mainly two variants to keep tracking, but I can't seem to integrate this with the hosted pretix:
* Load a pixel image from the tracking server
* Redirect back to our conference site once the ticket has been purchased

But markdown and html loading of images seems to be prohibited and I can't find a redirect option.

I see there is a campaign plugin, but it does not mention matomo and the price is beyond our budget.

Are there any other options?

**Comment by pc-coholic** [comment, marked as answer]
The plugin you're looking for is not the campaigns plugin (as it only does very limited, pretix-local campaign tracking with no external dependencies) - instead you need the "Tracking Codes" plugin.

For pretix.eu SaaS-customers, it can be found and activated in the Integrations-tab of the plugins-page.

Please note: you can only use the plugin with Matomo, if your organizer account or event are tied to a custom domain.

**Comment by dune73** [comment]
Thank you for the swift response @pc-coholic. I had looked into the "Tracking Codes" plugin too, but since it does not mention Matomo in the description I thought this to be a non-option. You maybe want to fix that.

And sorry: I do not understand the _if your organizer account or event are tied to a custom domain_. Our conference runs at example.com, then our users click on a link and land at pretix.eu/example where they select the ticket and complete the purchase. Does this fit the requirement?

And if not: Is there any other requirement? (I see the privacy aspect in all of this and I do not care about the individual user or their history. The only thing I want to know is (1) number of visitors on website (2) number of users who clicked the link to the shop (3a) number of users who selected a ticket (3b) number of users who completed the purchase.

**Reply by wiffbi** [reply]
You need to register a domain with your pretix Hosted account (either organizer or event). See help here: https://docs.pretix.eu/en/latest/user/organizers/domain.html#using-a-custom-domain

Ideally that domain is a subdomain of your main domain, e.g. when you have your marketing website on example.com (or www.example.com), register e.g. tickets.example.com with pretix as described above. That way, user sessions won’t be lost as tracking cookies work accross subdomains of the same main domain.

**Reply by dune73** [reply]
Thank you very much @wiffbi. This makes it clear.

**Comment by dune73** [comment]
Since the paid plugin is beyond what we are willing to pay and since we have not set this up with a subdomain, I thought about a custom setup that would give me the information I wanted.

I have been fetching the number of tickets sold every 5 min for a long time. I feed this into a local data file.

What I am doing now is I examine that counter and if it goes up, I make a custom curl call to matomo and trigger the goal counter. That event is then recorded. It is not linked to the session that exited towards pretix, but I am not interested in the session anyways and we sell relatively few tickets, so if I need to know, I could still correlate by hand.

This means I have a sales funnel with 3 events now:
* Access to website
* Exit via link to pretix site (goal 1)
* Ticket sale counter growing triggers custom curl call (goal 2) 

This is a bit of a hack apparently, but it solves my problem without having to change our pretix setup.
